time_slider_gazebo:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 1000  
  
  # Position Controllers ---------------------------------------
  left_hip_pitch_position_controller:
    type: effort_controllers/JointPositionController
    joint: Left_Pitch
    pid: {p: 12000.0, i: 0, d: 200.0}
  left_hip_roll_position_controller:
    type: effort_controllers/JointPositionController
    joint: Left_Roll
    pid: {p: 7000.0, i: 0, d: 150.0}
  left_hip_slide_position_controller:
    type: effort_controllers/JointPositionController
    joint: Left_Slide
    pid: {p: 12000.0, i: 0, d: 500.0}
  left_ankle_pitch_position_controller:
    type: effort_controllers/JointPositionController
    joint: Left_Foot_Pitch
    pid: {p: 6000.0, i: 0, d: 50.0}
  left_ankle_roll_position_controller:
    type: effort_controllers/JointPositionController
    joint: Left_Foot_Roll
    pid: {p: 6000.0, i: 0, d: 50.0}
  right_hip_pitch_position_controller:
    type: effort_controllers/JointPositionController
    joint: Right_Pitch
    pid: {p: 12000.0, i: 0, d: 200.0}
  right_hip_roll_position_controller:
    type: effort_controllers/JointPositionController
    joint: Right_Roll
    pid: {p: 7000.0, i: 0, d: 150.0}
  right_hip_slide_position_controller:
    type: effort_controllers/JointPositionController
    joint: Right_Slide
    pid: {p: 12000.0, i: 0, d: 500.0}
  right_ankle_pitch_position_controller:
    type: effort_controllers/JointPositionController
    joint: Right_Foot_Pitch
    pid: {p: 6000.0, i: 0, d: 50.0}
  right_ankle_roll_position_controller:
    type: effort_controllers/JointPositionController
    joint: Right_Foot_Roll
    pid: {p: 6000.0, i: 0, d: 50.0}

  # Torque Controllers ---------------------------------------
  left_hip_pitch_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Left_Pitch
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_hip_roll_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Left_Roll
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_hip_slide_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Left_Slide
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_ankle_pitch_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Left_Foot_Pitch
    pid: {p: 100.0, i: 0.01, d: 10.0}
  left_ankle_roll_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Left_Foot_Roll
    pid: {p: 100.0, i: 0.01, d: 10.0}
    
  right_hip_pitch_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Right_Pitch
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_hip_roll_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Right_Roll
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_hip_slide_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Right_Slide
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_ankle_pitch_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Right_Foot_Pitch
    pid: {p: 100.0, i: 0.01, d: 10.0}
  right_ankle_roll_torque_controller:
    type: effort_controllers/JointEffortController
    joint: Right_Foot_Roll
    pid: {p: 100.0, i: 0.01, d: 10.0}
